### ISSUE TITLE
Allow passing file content from stdin

### ISSUE THREAD

**Issue by InSyncWithFoo** [microsoft/pyright#7282]
**Problem**

I'm writing a Pyright integrating plugin for PyCharm. Currently the Pyright CLI doesn't allow passing a file via stdin, which means I would have to programmatically save the file first. This is <em>very</em> cumbersome and in some cases most undesirable.

Using the LSP directly is not a choice, since [the community version](https://plugins.jetbrains.com/docs/intellij/language-server-protocol.html#supported-ides) doesn't support custom LSPs.

**Suggestion**

Something like this would be nice:

```shell
$ echo "def foo() -> int: return 'Hello world!'" | pyright --via-stdin --stdin-filename foo.py
```

```none
foo.py
  foo.py:1:26 - error: Expression of type "Literal['Hello world!']" cannot be assigned to return type "int"
    "Literal['Hello world!']" is incompatible with "int" (reportReturnType)
1 error, 0 warnings, 0 informations 
```

**Comment by erictraut** [comment]
I'm going to reject this because I don't think this is well motivated. 

You should use the LSP interface if you want to integrate pyright into an editor. The command-line version is not designed for this use case. Please use an editor that supports LSP.
